### PR TITLE
Revamp chain & exchange clients with TLS & sticky session to support k8s, lb endpoints

### PR DESCRIPTION
**Pull request by nhannamsiu**
Effort to align sdk-go structure with sdk-python's to provide better, consistent experience to API users.
The chain client init process is a bit different here as it makes use of default cosmos-sdk/tm client, while we craft everything from scratch on our sdk-python.

**Changes**
* Use default broadcast methods with our own txClient instead of cosmos-sdk client
* Sticky session for chain, exchange client to pair with load balancing endpoints
* Abstract away direct proto binding usage
* Examples for chain & exchange clients' usage
* Up-to-date proto binding from injective-core, injective-exchange
* Convenient network class to quickly select and test an endpoint in a specific network

### CODE REVIEW

**Review by xlab — APPROVED** [review]
I am worried that endpoints are hardcoded in the code. Not defined by env config.
Maybe I'm missing the place where a dynamic configuration can be applied. For 3rd party deployments.

File: client/chain.go
Comment (xlab) [inline client/chain.go]: I don't like that the batching code is being scrapped completely. It was added on purpose, resolving the issues in the default mechanism provided by Cosmos SDK, which doesn't support more throughput than 1 tx per block. Can you please document the rationale that is going behind these changes? Including all the drawbacks that this may impose.

Also, what was the original issue?

**Review by xlab — CHANGES_REQUESTED** [review]

File: client/chain.go
Comment (xlab) [reply client/chain.go]: The sdk-go client has been built to be bullet-proof and was thoroughly used for many months now, what's the particular reason to change it?

**Comment by nhannamsiu** [comment]
> I am worried that endpoints are hardcoded in the code. Not defined by env config.
This is sdk, env loading should be done by the application. MM, API users use the sdk to craft their own usage and there are probably more custom env vars on their app.
The `network.go` is simply a util class abstracting some init steps, offer all endpoints we have and let users run these examples without configuring anything.

> I don't like that the batching code is being scrapped completely.
This is a timely PR to get the TLS, sticky session working for sdk-go so it unblocks Powen to use load balancing endpoint. Right now he still has to manually switching sentries.
It's also simpler to get the default broadcast methods ready 1st with cookie session, I'm going to add it back soon. It's great cause it makes tx broadcasting efficient.
About the limit 1 tx / block. It's also resolved with the default sync, async clients I added.

> Also, what was the original issue?
No issue at all, just to get this PR up asap. Also, the MM tends to react swiftly when it comes to placing, canceling orders. The batch mode is efficient for non-critical msgs and wound't affect the simple clients, so we can add it back after you review the current impl for sticky session & new dir structure.

**Comment by xlab** [comment]
> It's also simpler to get the default broadcast methods

Nothing should be forced into prod just to unblock someone, especially dumbing out quite advanced logic that was there for a reason.

> The batch mode is efficient for non-critical msgs 

My concern is not only about the batch mode, but also about how easily we introduce "default" methods there which are broken by design, for example, [BlockBroadcastTx](https://github.com/cosmos/cosmos-sdk/blob/ccaf94391e1b46696df56d0395b5e6acc46b850d/client/broadcast.go#L87)

As I mentioned in the previous message, the code in this part has been ironed out of bugs during the past year, and it had lots of problems that we had to catch and debug. Let's not touch something that works well now, especially making an assumption that "default" implementation from Cosmos SDK is worthy.

**Comment by nhannamsiu** [comment]
0. Nothing is forced. They are official methods to broadcast and they work.
1. `BlockBroadcastTx` works as it's designated to be: wait till the tx is included in block response to return, aka 1 tx / block.
Sync and Async allow sending as many txs as one needs within 1 block.
2.  10s batch flush is no good for MM, they want to act asap when something changed. I've been working closely with trading team and know what they need. Upon this timely revamp, it's sufficient for them to get started.
3. As I said, the batch broadcast will be added again after you review current PR. Let's see if they like 10s batch flush to handle ~2.5s block time with 10+ msgs or txs per block.

**Comment by xlab** [comment]
0-1 : I am saying that those are broken because we spent a great amount of time in the past to fix the issues and the resulting code that you saw in the package was born to overcome all the issues. The official documentation of cosmos asks to avoid using `BlockBroadcastTx`, and if you check out the internal logic of the handler, you'll see why.

2-3: `10s batch flush is no good` - there is no 10s batch flush, you're misreading the code. It's `500ms`. Besides that, there is no need to use Batching method at all, if MMs are constructing batches of messages themselves.

**Comment by nhannamsiu** [comment]
Ah yes you're right, 500ms batch flush and 100ms status polling. I took a quick glance and crossed that.
Block broadcast is bad, sync/async should be the ones to use.
I will add back the batch broadcast back and request review.

**Review by xlab — APPROVED** [review]

**Pull request merged**
